shuffler/logic/Dungeons/Courage Temple/CourageTemple.logic

```diff
@@ -94,41 +125,67 @@ area CourageTemple:
     TopRightLockedDoor -> TopRight: open CourageKey
     TopRightLockedDoor -> StairsToF2EmptyRoom: open CourageKey
 
+
   room B1:
     node MiddleMovingPlatform:
       door ToF1Arena
       door ToF1PolsVoice
 
-    node IslandNearMovingPlatform:
-      chest IslandChest
+    node ColorSwitchBottom: # Leaving this point, we can be at any of the two state
+      gain BlueSwitch
+      gain RedSwitch
+
+    MiddleMovingPlatform <-> ColorSwitchBottom: (item Bow | item Boomerang | item Hammer)
 
     node InvisibleMaze:
       door ToF1InvisibleMaze
       chest BowChest
+      chest IslandChest
 
     node LeftmostStairsB1:
       door ToF1Leftmost
 
-    MiddleMovingPlatform -> IslandNearMovingPlatform: item GrapplingHook
+    node ColorSwitchLeft:
+      gain BlueSwitch
+      gain RedSwitch
+
+    LeftmostStairsB1 <-> ColorSwitchLeft: item Bow
+
+    MiddleMovingPlatform -> InvisibleMaze: item GrapplingHook
 
     InvisibleMaze -> LeftmostStairsB1: item Bow
-    InvisibleMaze <-> IslandNearMovingPlatform
+
 
     node Torches:
-      door F1Torches
+      door ToF1Torches
+
+    node AfterTorches
 
-    node AfterTorches:
+    Torches -> AfterTorches: item Boomerang
+
+    node Target:
       chest EyeTarget
-      door F1TopRight
 
-    Torches -> AfterTorches: (item Bow & item Boomerang)
+    node ColorSwitchRight:
+      gain BlueSwitch
+      gain RedSwitch
+
+    AfterTorches <-> Target: (item Bow & item Boomerang)
+    AfterTorches <-> ColorSwitchRight: item Bow
```

Comment:
I think this one requires the boomerang as well. The arrow shooter thing starts off aimed to the right, when it needs to be changed to left to hit the color switch.

Reply:
You are correct, I will commit the change